msal/src/main/java/com/microsoft/identity/client/internal/controllers/LocalMSALController.java

```diff
@@ -283,7 +285,7 @@ public AcquireTokenResult acquireTokenSilent(
 
         final List<ICacheRecord> cacheRecords = tokenCache.loadWithAggregatedAccountData(
                 parametersWithScopes.getClientId(),
-                TextUtils.join(" ", parametersWithScopes.getScopes()),
+                TextUtils.join(" ", requestScopes),// use only the requested scopes when retrieving access token.
```

Comment:
Not clear to me why this is being done? Looks we won't get a new ID Token now?

Reply:
Okay - this is not token request. Just cache lookup

Reply:
This is when we are retrieving the `access token` - we match if the scopes exists in the stored access token, so we should not include the `default` or the `oidc` scopes.

We don't match `idtoken` on scopes when retrieving

Reply:
But do we need the default scopes for lookup? Looks like we used to do so

Reply:
Okay we omit default scopes at the lower cache lookup layer

Reply:
We may want to keep it that way, your call

Reply:
We always omit `default scopes` while [comparing here](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/cache/AbstractAccountCredentialCache.java#L245). In fact , we never save default scopes for accesstoken too , see [here ](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/cache/MicrosoftStsAccountCredentialAdapter.java#L142). So it's essentially a no-op, so we better only compare with requested scopes as only they will be present in the AT.